File: runtime/GenAI.ts
declare class ChatMessage {
    Actor: string
    Body: string
}

class _GenAI {
    private native: any;

    Generate(systemPrompt: string, prompt: string): string {
        return this.native.Generate(systemPrompt, prompt);
    }
    Chat(systemPrompt: string, history: ChatMessage[], actor: string, pose: string): ChatMessage[] {
        return this.native.Chat(systemPrompt, history, actor, pose);
    }
}

declare const GenAI: _GenAI

File: internal/genai/GenAI.go
package genai

type GenAI interface {
	Generate(systemPrompt string, prompt string) string
	Chat(systemPrompt string, history []ChatMessage, actor, pose string) []ChatMessage
}

type ChatMessage struct {
	Actor string
	Body  string
}

func NewGenAI(address string, model string) (GenAI, error) {
	return NewOllamaGenAI(address, model)
}
